### ISSUE TITLE
Warn about misuse of `import-style` rule

### ISSUE THREAD

**Issue by FloEdelmann** [sindresorhus/eslint-plugin-unicorn#924]
Follow-up to #899 and #910:

> […] Configurations like below in my opinion present a misuse of this rule.
> 
> ```js
>       fs: {
>         unassigned: false,
>         default: false,
>         namespace: false,
>         named: false,
>       },
> ```
> 
> I think it should be simply disallowed and `no-restricted-imports` or `no-restricted-modules` rules should be recommended to the user when such configuration is detected.

_Originally posted by @futpib in https://github.com/sindresorhus/eslint-plugin-unicorn/issues/899#issuecomment-735985491_

**Comment by fisker** [comment]
PR welcome.
